lib/rock/webapp/tasks/api.rb

```diff
@@ -250,29 +250,49 @@ def get_operation(name_service, name, operation_name)
                         end
                     end
 
-                    desc 'connect a port, /connect&to=taskname&port=portname, optional &type=buffer&size=10'
+                    desc 'connect a port, /connect&to=hostname/taskname&port=portname&host=hostname, optional &type=buffer&size=10'
                     params do
                         requires :to, :port
                         optional :type, type: String, default: "data"
                         optional :size, type: Integer, default: 10
                     end
                     get ':name_service/:name/ports/:port_name/connect', requirements: { name_service: ValidHostnameRegex } do
-                        target = port_by_task_and_name(*params.values_at('name_service', 'to', 'port'))
+                        hostname = "*"
+                        #check if "to" contains '/'
+                        taskinfo = params["to"].split('/')
+                        if taskinfo.size() == 2
+                           #there was a / present
+                           hostname=taskinfo[0]
+                           taskname=taskinfo[1]
+                        elsif taskinfo.size == 1
+                           taskname = taskinfo[0]
+                        end
+                        target = port_by_task_and_name(hostname, taskname, params["port"])
                         source = port_by_task_and_name(*params.values_at('name_service', 'name', 'port_name'))                        
                         if request.params["type"] == "buffer"
-                            source.connect_to target, :type => :buffer, :size => request.params["size"]
+                            res = source.connect_to target, :type => :buffer, :size => request.params["size"]
                         else
-                            source.connect_to target
+                            res =source.connect_to target
```

Comment:
orocos.rb raises on error, never returns a boolean. You'll need to catch and report a proper error.